pkg/e2e/operators/rbac.go

```diff
@@ -0,0 +1,122 @@
+	checkUpgrade(helper.New(), &operatorv1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rbac-permissions-operator",
+			Namespace: "openshift-rbac-permissions",
+		},
+		Spec: &operatorv1.SubscriptionSpec{
+			Package:                "rbac-permissions-operator",
+			Channel:                getChannel(),
+			CatalogSourceNamespace: "openshift-rbac-permissions",
+			CatalogSource:          "openshift-rbac-permissions",
+		},
+	})
+})
+
+func checkSubjectPermissions(h *helper.H, spName string) {
+	ginkgo.Context("SubjectPermission", func() {
+		ginkgo.It("should have the expected ClusterRoles, ClusterRoleBindings and RoleBindinsg", func() {
+			clusterRoles, clusterRoleBindings, roleBindings, err := getSubjectPermissionRBACInfo(h, spName)
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, clusterRoleName := range clusterRoles {
+				_, err := h.Kube().RbacV1().ClusterRoles().Get(clusterRoleName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred(), "failed to get clusterRole %v\n", clusterRoleName)
+			}
+
+			for _, clusterRoleBindingName := range clusterRoleBindings {
+				err := pollClusterRoleBinding(h, clusterRoleBindingName)
+				Expect(err).ToNot(HaveOccurred(), "failed to get clusterRoleBinding %v\n", clusterRoleBindingName)
+			}
+			for _, roleBindingName := range roleBindings {
+				err := pollRoleBinding(h, h.CurrentProject(), roleBindingName)
+				Expect(err).NotTo(HaveOccurred(), "failed to get roleBinding %v\n", roleBindingName)
+			}
+
+		}, float64(float64(config.Instance.Tests.PollingTimeout)))
+	})
+}
+
+func getSubjectPermissionRBACInfo(h *helper.H, spName string) ([]string, []string, []string, error) {
+	us, err := h.Dynamic().Resource(schema.GroupVersionResource{
+		Group:    "managed.openshift.io",
+		Version:  "v1alpha1",
+		Resource: "subjectpermissions"}).Namespace(operatorNamespace).Get(spName, metav1.GetOptions{})
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("Error getting %s SubjectPermission", spName)
+	}
+
+	subjectName, _, err := unstruct.NestedString(us.Object, "spec", "subjectName")
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("Error getting subjectName")
+	}
+
+	clusterRoles, _, err := unstruct.NestedStringSlice(us.Object, "spec", "clusterPermissions")
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("Error getting clusterPermissions")
+	}
+
+	var clusterRoleBindings = []string{}
+	for _, crName := range clusterRoles {
+		clusterRoleBindings = append(clusterRoleBindings, crName+"-"+subjectName)
+	}
+
+	permissions, _, err := unstruct.NestedSlice(us.Object, "spec", "permissions")
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("Error getting permissions")
+	}
+
+	var roleBindings = []string{}
+	for _, p := range permissions {
+		perm := p.(map[string]interface{})
+		cpName, _, err := unstruct.NestedString(perm, "clusterRoleName")
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("Error getting permission clusterRoleName", cpName)
```

Comment:
Looks like this is the cause of the test failure.

Reply:
Thanks @meowfaceman!

Updated now :+1: